### DISCUSSION TITLE
Unable to interact with cypress recordings during workflow

### DISCUSSION THREAD

**Discussion by RomanoTorri** [cypress-io/cypress#30459] [Questions and Help]
I want to upload the video taken by cypress to a separate service after the spec has finished. I've tried putting the code to do this in a node event (after:spec), however, the video has not finished processing even at this point so it just uploads a corrupted 9 byte .MP4.

Is it possible to interact with the finished video file in the cypress workflow at all? I run multiple specs in one run so the after:run node event is not appropriate either.

**Comment by portyu9** [comment]
Hi @RomanoTorri 

This is expected timing for `after:spec`.

Cypress fires `after:spec` **before it proceeds with processing the spec's video**. In fact, if your `after:spec` handler returns a Promise, Cypress waits for that Promise before it starts/continues the video-processing step. That is why the file can exist but still be an unusable tiny MP4 at that point.

There is no documented per-spec `after:video` event that fires after final encoding.

The supported options are:

1. Upload finalized videos in `after:run` or in the CI step after `cypress run` exits.
2. If each video's upload must happen immediately after its spec, run each spec in its own Cypress process and upload the video after that process exits.

I would avoid polling the file until its size stops changing; it is inherently racy and depends on Cypress's internal encoding timing.

Please mark this answer as accepted if it helped, thank you.
